### PR TITLE
nightlies: Provide script to generate meta-information

### PR DESCRIPTION
**Pull request by miri64**
This provides a script that parses all meta-information I could find and dumps them in a JSON file `/RIOT-OS/RIOT/nightlies` in a format not unsimilar to the `/api/pull_requests` method of Murdock. This is preparation for the a graphical representation of the nightlies at https://ci.riot-os.org.

### CODE REVIEW

File: update_nightly_list.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+# vim:fenc=utf-8
+#
+# Copyright © 2018 Martine Lenders <m.lenders@fu-berlin.de>
+#
+# Distributed under terms of the MIT license.
```
Comment (miri64) [inline update_nightly_list.py]: Oops, did not adapt my vim templates for this Repo. Will fix license and copyright notice (and all the python2 stuff) tomorrow.

File: update_nightly_list.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if branch not in nightlies:
+            # if branch is not yet in the nightlies list, create an instance
+            nightlies[branch] = []
+        # get hash of latest commit from symlink
+        latest_commit = os.path.realpath(os.path.join(branch_dir, "latest"))\
+            .split(os.path.sep)[-1]
+        # check if it was already build in a previous build
+        build = find_commit_build(nightlies[branch], latest_commit)
+        if build is None:
+            build = {"commit": latest_commit}
```
Comment (miri64) [inline update_nightly_list.py]: I actually like to also have the commit summary and author here, but I wasn't sure where (if anywhere) to access the git repository.

Comment (miri64) [reply update_nightly_list.py]: Same goes for the `output_url`, but I know where I can get that from (I just forgot about it)

**Comment by miri64** [comment]
Ping @kaspar030?

File: update_nightly_list.py (hunk 1 above)
Comment (kaspar030) [reply update_nightly_list.py]: this is done, right?

File: update_nightly_list.py (hunk 2 above)
Comment (kaspar030) [reply update_nightly_list.py]: Murdock does have that information, maybe we can make it export those.

File: update_nightly_list.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+# Distributed under terms of the MIT license.
+
+import argparse
+import datetime
+import json
+import os
+import re
+import time
+
+FILENAME = "nightlies"
```
Comment (kaspar030) [inline update_nightly_list.py]: maybe "nightlies,json"?

File: update_nightly_list.py (hunk 1 above)
Comment (miri64) [reply update_nightly_list.py]: Nope, totally forgot about that... sorry

File: update_nightly_list.py (hunk 2 above)
Comment (miri64) [reply update_nightly_list.py]: Where are they?

**Review by kaspar030 — CHANGES_REQUESTED** [review]

File: update_nightly_list.py (hunk 2 above)
Comment (kaspar030) [reply update_nightly_list.py]: Murdock exports quite some info via environment: 
https://github.com/kaspar030/murdock:/blob/d2655fe7f7204374ec7e65810d89e3e3aba8c0e6/murdock.py#L214

The PR author github name should already be available as ```CI_PULL_USER```

Here would be the place to get more fields from the PR json:
https://github.com/kaspar030/murdock/blob/d2655fe7f7204374ec7e65810d89e3e3aba8c0e6/murdock.py#L274

**Comment by kaspar030** [comment]
Looks good. I was able to manually generate the nightlies file.
Just two things:

1. any reason you're not using nightlies.json?

2. maybe it makes sense to only handle one branch at a time? It would simplify the script, and there'd be not race conditions, ... I mean, just use <branch>/nightlies.json for each branch.

**Comment by miri64** [comment]
> maybe it makes sense to only handle one branch at a time? It would simplify the script, and there'd be not race conditions, ... I mean, just use /nightlies.json for each branch.

This would make the front-end script more complicated, because it has to know which branch are build, but okay. Maybe we can add some information on that later.

**Review by kaspar030 — APPROVED** [review]

**Comment by kaspar030** [comment]
ACK.

**Pull request merged**